### ISSUE TITLE
Possible issue with summary.queue_list when calling queuecomputer in shiny

### ISSUE THREAD

**Issue by MarkHanly** [AnthonyEbert/queuecomputer#34]
Hi @AnthonyEbert,

I've really been enjoying woking with queuecomputer, thank you for a great package!

I would like to develop a shiny app as an interface to the work I have been doing but seem to be running in to a problem when calling `summary()` on a `queue_list` object in shiny. Apologies if advance this is a shiny issue and not a queuecomputer issue. 

The error returned at the console is `Warning: Error in summary.queue_list: object 'utilization' not found`. Trying to isolate the problem, it seems to only arise when there is an option to dynamically set the number of servers via a shiny widget. The problem goes away if the number of servers is fixed. Widgets to change the arrival and service parameters within shiny work as expected.

As a quick demonstration of the unexpected and expected behaviour, I've implemented two simple queues in shiny with and without the option to dynamically set the number of servers here: https://cbdrh.shinyapps.io/shinyQueueError/. (This app will go down at some point in the future but for posterity, or to reproduce locally, the code is here: https://github.com/MarkHanly/shinyQueueError).

Below is a simple shiny app.R file that gives a MWE replication of the problem. Switching out the commented code so that the number of servers is fixed shows the app working as expected.

Thank you again for a great package, and any suggestions on this unexpected behaviour.

All the best,
Mark

**MWE code to reproduce issue (app.R):**

```
library(queuecomputer)
library(shiny)

#### Define UI ####
ui <- fluidPage(

    # Application title
    titlePanel("Testing queuecomputer in shiny"),

    # Show a plot of the generated distribution
    mainPanel(
       sliderInput('servers1', 'Number of servers', min = 1, max = 10, value = 3, step = 1),
       textOutput("test1")
    )

    )

#### Define server logic #### 

server <- function(input, output) {

    # A reactive queue_list that updates when the user changes the number of servers     
    
    # This throws an error
    queue1 <- reactive({
        a = cumsum(rexp(100, 1)) # arrival times
        s = rexp(100, 1) # service times
        # q <- queuecomputer::queue_step(arrivals = a, service = s, servers = 2) #This works as expected
        q <- queuecomputer::queue_step(arrivals = a, service = s, servers = input$servers1) 
        return(q)
    })    
    

    # The output to report: the utilisation factor returned by summary.queue_list    
    output$test1 <- renderText({
        x <- summary(queue1())
        paste('utilisation factor =', round(x$utilization, digits = 3))
    })
    
}

# Run the application 
shinyApp(ui = ui, server = server)

```

**Comment by AnthonyEbert** [comment]
Hi @MarkHanly  

Great to hear you like it! 

Thanks for making this easy for me. Turns out it's just a stupid bug in queuecomputer where if the servers input was of class integer then the utilisation wasn't calculated. I forked your shiny repository and put in a bandaid solution https://github.com/AnthonyEbert/shinyQueueError. I also put out a fix for queuecomputer (it's now 1.0.1) on github (not on CRAN yet) which completely fixes this problem. 

Thanks for raising this issue!

Cheers

Anthony
